pandas/io/parsers/base_parser.py

```diff
@@ -171,8 +174,14 @@ def __init__(self, kwds):
             if self.index_col is not None:
                 is_sequence = isinstance(self.index_col, (list, tuple, np.ndarray))
                 if not (
-                    is_sequence
-                    and all(map(is_integer, self.index_col))
+                    # error: Argument 2 to "map" has incompatible type
+                    # "Union[int, Sequence[int]]"; expected "Iterable[int]"
+                    (
+                        is_sequence
```

Comment:
could inline the instance check to remove this ignore, only used once.

Reply:
will do